tests/UnitTests/TemplateSource/TagTests/PluginModifier/PluginModifierNumberFormatTest.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace UnitTests\TemplateSource\TagTests\PluginModifier;
+
+/**
+ * class for modifier tests
+ *
+ * @runTestsInSeparateProcess
+ * @preserveGlobalState disabled
+ * @backupStaticAttributes enabled
+ */
+class PluginModifierNumberFormatTest extends \PHPUnit_Smarty
+{
+    public function setUp(): void
+    {
+        $this->setUpSmarty(__DIR__);
+    }
+
+    /**
+     * @return void
+     * @throws \SmartyException
+     *
+     * @dataProvider numberFormatDataProvider
+     */
+    public function testNumberFormat($template, $subject, $expectedString)
+    {
+        $this->smarty->assign('subject', $subject);
+
+        $tpl = $this->smarty->createTemplate($template);
+
+        $this->assertEquals($expectedString, $this->smarty->fetch($tpl));
+    }
+
+    public function numberFormatDataProvider()
+    {
+        return [
+            'default'  => [
+                'template'      => 'string:{$subject|number_format}',
+                'subject'       => 12345,
+                'expectedString' => "12,345",
+            ],
+            'withDecimalDefault'  => [
+                'template'      => 'string:{$subject|number_format}',
+                'subject'       => 12345.6789,
+                'expectedString' => "12,346",
+            ],
+            'withDecimalAndExtras'  => [
+                'template'      => 'string:{$subject|number_format:2:\'-\':\'~\'}',
```

Comment:
You could use double quotes inside singles to avoid unnecessary quoting.

Reply:
Updated this to use double quotes